core/src/test/java/org/elasticsearch/threadpool/GenericThreadPoolTests.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.threadpool;
+
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.common.util.concurrent.EsThreadPoolExecutor;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasToString;
+
+public class GenericThreadPoolTests extends ESThreadPoolTestCase {
+
+    public void testGenericThreadPoolConfiguration() throws InterruptedException {
+        final Settings.Builder builder = Settings.builder();
+        final int max;
+        if (randomBoolean()) {
+            final int processors = randomIntBetween(1, 64);
+            builder.put("processors", processors);
+            max = ThreadPool.boundedBy(4 * EsExecutors.boundedNumberOfProcessors(builder.build()), 128, 512);
```

Comment:
can we put a static method in ThreadPool that does this calculation and that is shard between production code and tests? We use these magic constants in multiple places in the tests here.